### ISSUE TITLE
[Bug] repeated loading when use fuzzydir.lua

### ISSUE THREAD

**Issue by dyphire** [sibwaf/mpv-scripts#3]
As I mentioned there: https://github.com/sibwaf/mpv-scripts/pull/2

For the following folder structure:
```
│   1.mkv
│   1.srt
└───anyname*
    │   1_en.srt
    └───anyname*
        │   1_zh.srt
```
Before use the script of fuzzydir.lua: mpv only had one 1.srt.
After use the script of fuzzydir.lua: mpv had two 1.srt.

The same is true for audio files.

As in the example, this issue only happen in the same directory of the video file. Normal loading in subdirectory.
I think the same file should only be loaded once on mpv.

**Comment by sibwaf** [comment]
This script was made to add path discovery support to mpv, not to select the correct file for the user. It is also not a bug as far as I'm concerned, as the "vanilla" mpv has the same behavior.

Config (the script is NOT loaded):
`sub-file-paths=subs;subs/moresubs` (for Windows; replace the semicolon with a colon on other platforms)

Directory structure:
```
|  video.mkv
|  video_1.ass
|  subs
   |  video_2.ass
   |  moresubs
      |  video_3.ass
```

mpv's console output:
```
     Subs  --sid=3 '[Beatrice-Raws] Byousoku 5 Centimeter [BDRip 1920x1080 x264 FLAC]_1.ass' (ass) (external)
     Subs  --sid=4 '[Beatrice-Raws] Byousoku 5 Centimeter [BDRip 1920x1080 x264 FLAC]_2.ass' (ass) (external)
     Subs  --sid=5 '[Beatrice-Raws] Byousoku 5 Centimeter [BDRip 1920x1080 x264 FLAC]_3.ass' (ass) (external)
```

The "only one file was loaded" behavior before is probably caused by a non-configured or misconfigured `sub-file-paths`.

**Comment by dyphire** [comment]
You understand wrong. I don't mean that mpv will only load one file. Instead, the same file is loaded only once.

For your example, using the script of fuzzydir.lua will get the following results:
```
     Subs  --sid=3 '[Beatrice-Raws] Byousoku 5 Centimeter [BDRip 1920x1080 x264 FLAC]_1.ass' (ass) (external)
     Subs  --sid=4 '[Beatrice-Raws] Byousoku 5 Centimeter [BDRip 1920x1080 x264 FLAC]_1.ass' (ass) (external)
     Subs  --sid=5 '[Beatrice-Raws] Byousoku 5 Centimeter [BDRip 1920x1080 x264 FLAC]_2.ass' (ass) (external)
     Subs  --sid=6 '[Beatrice-Raws] Byousoku 5 Centimeter [BDRip 1920x1080 x264 FLAC]_3.ass' (ass) (external)
```

**Comment by sibwaf** [comment]
Yep, managed to reproduce the bug. I'll take a look, thanks for the report!

**Comment by dyphire** [comment]
not use fuzzydir.lua
![](https://cdn.jsdelivr.net/gh/dyphire/PicGo/img/2022/04/05/20220405020616.png)
use fuzzydir.lua
![](https://cdn.jsdelivr.net/gh/dyphire/PicGo/img/2022/04/05/20220405020553.png)
Now you can see the problem of expected behavior more intuitively. The script is repeatedly loading the same file in the same directory of the video. Again, subdirectories are expected to behave normally.

**Comment by dyphire** [comment]
The latest commit has fixed this issue and the script is now working properly.

**Comment by dyphire** [comment]
> Maybe there should be a threshold to limit the amount of discovered paths or something.

I also want to know another thing. Will this be realized?

**Comment by sibwaf** [comment]
Hm, I'm not sure to be honest. Feels a bit too hackish, so I'd like to debug it before making any decisions. Could you share any info on how to reproduce the crash (in a new issue)? `max_search_depth`, `sub-file-paths` and directory structure should probably be enough.

**Comment by dyphire** [comment]
> Could you share any info on how to reproduce the crash (in a new issue)?

Sorry, I don't have the conditions to reproduce the relevant issues.

This issue was reported to me by a friend of mine. He plays files through SMB protocol, He gets a very bad delayed loading experience when uses this script with thousands of files in the video directory. 

It can be said that this problem is actually caused by poor document management habits.
Therefore, it is understandable even if the relevant optimization is not implemented.

**Comment by sibwaf** [comment]
Well, either way just having the script cause problems isn't good, there should be a fail-safe of sorts. Made #4, I'll look into it someday.
